packages/react-storage/src/components/StorageBrowser/Views/Controls/Table.tsx

```diff
@@ -10,5 +147,101 @@ export interface TableControl<T extends StorageBrowserElements>
     | 'TableHeader'
     | 'TableRow'
   > {
-  (): React.JSX.Element;
+  <U extends ILocation>(props: TableControlProps<U>): React.JSX.Element;
+  SortIndeterminateIcon: T['Icon'];
+  SortAscendingIcon: T['Icon'];
+  SortDescendingIcon: T['Icon'];
 }
+
+const isSortableTableControlProps = <T extends ILocation>(
+  props: TableControlProps<T>
+): props is SortableTableControlProps<T> => {
+  return 'onSort' in props && 'sortState' in props;
+};
+
+const sortAriaLabel = {
+  none: 'not sorted',
+  ascending: 'sorted ascending',
+  descending: 'sorted descending',
+};
+
+export const TableControl: TableControl = <U extends ILocation>(
+  props: TableControlProps<U>
+) => {
+  const { data, ariaLabel } = props;
+  const { rows, columns } = data;
+
+  let onSort: (columnKey: Column<U>) => void;
+  let sortState: { key: keyof U; direction: SortDirection };
+
+  if (isSortableTableControlProps(props)) {
+    ({ onSort, sortState } = props);
+  }
+
+  return (
+    <Table aria-label={ariaLabel}>
+      <TableHead>
+        <TableRow>
+          {columns?.map((column) => (
+            <TableHeader
+              key={column.key as string}
+              aria-sort={
+                column.key === sortState?.key ? sortState.direction : 'none'
+              }
+            >
+              {column.sortable && onSort ? (
+                <TableHeaderButton
+                  aria-label={`${column.header}, ${
+                    sortState.key === column.key
+                      ? sortAriaLabel[sortState.direction]
+                      : sortAriaLabel['none']
+                  }`}
```

Comment:
I originally thought `aria-sort` in the `th` element would be announced when it changed, but that wasn't the case. As a result I added this aria label to get read by the screen reader when you press the button. With VoiceOver, I was able to press the button with `ctrl+option+space` and have it say, `Name, sorted ascending` for instance

When moving through the table cells, it also reads this out which I thought was nice, but open to feedback on it